### PR TITLE
BuildPharPackage action

### PR DESCRIPTION
**Pull request by ralf57**
Note: do not merge yet

### CODE REVIEW

**Comment by jeanmonod** [comment]
Nice task! Seems to be almost good for merge? There is just the todo about stub...
The only thing you can maybe add is a functional test. I't pretty easy, take a look at the example in https://github.com/liip/RMT/tree/master/test/Liip/RMT/Tests/Functional

Anyway thanks for your work!

**Comment by ralf57** [comment]
Thanks for this first review.
I planned to do some refinements today
* Stub
* Metadata
* Tests
Will ping you when done.

**Comment by acrobat** [comment]
Nice feature @ralf57!

**Comment by ralf57** [comment]
@jeanmonod please have a final review after the build finished

File: src/Liip/RMT/Action/BuildPharPackageAction.php [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+    protected $releaseVersion;
+
+    public function __construct($options)
+    {
+        $this->options = array_merge(array(
+            'package-name' => 'rmt-package',
+            'destination' => '/tmp/',
+            'excluded-paths' => '',
+            'metadata' => [],
```
Comment (jeanmonod) [inline src/Liip/RMT/Action/BuildPharPackageAction.php]: This is not compatible with 5.3 ;)

Comment (ralf57) [inline src/Liip/RMT/Action/BuildPharPackageAction.php]: You're right.
But what about upping to PHP 5.4, especially considering http://php.net/supported-versions.php

Comment (jeanmonod) [inline src/Liip/RMT/Action/BuildPharPackageAction.php]: Personally I use it on two legacy projects that are still on 5.3... So as
long as it's not a big effort to keep that compatibility I prefer to keep it

Comment (ralf57) [inline src/Liip/RMT/Action/BuildPharPackageAction.php]: Fine for me and I'll push the quick fix.
BTW, the project will need to move to PHP 5.4 sooner or later but I'll leave that to you ;-)

File: src/Liip/RMT/Action/BuildPharPackageAction.php [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+    public function __construct($options)
+    {
+        $this->options = array_merge(array(
+            'package-name' => 'rmt-package',
+            'destination' => '/tmp/',
+            'excluded-paths' => '',
+            'metadata' => array(),
+            'default-stub-cli' => '<?php __HALT_COMPILER(); ?>',
+            'default-stub-web' => '<?php __HALT_COMPILER(); ?>',
```
Comment (jeanmonod) [inline src/Liip/RMT/Action/BuildPharPackageAction.php]: I just discover this function. But even with the doc: http://php.net/manual/en/function.halt-compiler.php I do not understand what this is doing here...

**Comment by jeanmonod** [comment]
Except my last comment, everything seems fine to be merged!

File: src/Liip/RMT/Action/BuildPharPackageAction.php (hunk 2 above)
Comment (ralf57) [inline src/Liip/RMT/Action/BuildPharPackageAction.php]: AFAIK, the __HALT_COMPILER() call is the minimal content for the stub. That's why I added it there. I also compared the contents of some generated stubs and the __HALT_COMPILER() call is always the last instruction.
Regarding its function, I agree the docs are not that clear, but it seems to do just that: halting the compiler during the package compilation.

Comment (jeanmonod) [inline src/Liip/RMT/Action/BuildPharPackageAction.php]: OK, I trust you ;)

**Comment by jeanmonod** [comment]
Will take some time this evening to merge and release this! Thanks

**Comment by ppetermann** [comment]
just to enlighten a bit about the __halt_compiler() function

@ralf57 is right about that, see http://php.net/manual/en/phar.fileformat.stub.php

also the reasoning is quite simple,
the stub in a phar file gets executed when you either run php file.phar or when you include file.phar,
the __halt_compiler() function basically tells php "stop interpreting the rest of the file", which is necessary as the rest that follows is the manifest, the contents and a signature, where manifest and signature are no valid php code, and contents doesn't need to be.

**Comment by jeanmonod** [comment]
@ppetermann Thank you for the enlightening!

**Comment by jeanmonod** [comment]
Merges and released! Thank you @ralf57 for this nice addition!

**Pull request merged**